packages/angular_devkit/build_angular/src/differential-loading/create-webpack-multi-config.ts

```diff
@@ -0,0 +1,185 @@
+
+
+import { AngularCompilerPlugin } from '@ngtools/webpack';
+import * as webpack from 'webpack';
+
+const StatsPlugin = require('stats-webpack-plugin');
+
+
+export function createWebpackMultiConfig(config: webpack.Configuration): webpack.Configuration[] {
+
+  const legacyConfig: webpack.Configuration = {
+    ...config,
+    output: buildLegacyOutput(config),
+    plugins: buildLegacyPlugins(config),
+    resolve: buildLegacyResolve(config),
+  };
+
+  const modernConfig: webpack.Configuration = {
+    ...config,
+    output: buildModernOutput(config),
+    plugins: buildModernPlugins(config),
+    entry: buildModernEntry(config),
+    resolve: buildModernResolve(config),
+  };
+
+  const newConfig = [legacyConfig, modernConfig];
+
+  return newConfig;
+}
+
+function buildLegacyOutput(config: webpack.Configuration): webpack.Output | undefined {
+
+  if (!config.output || !config.output.filename) {
+    return config.output;
+  }
+
+  const filename = config.output.filename.replace('[name]', '[name].legacy');
+
+  return { ...config.output, filename };
+}
+
+function buildModernOutput(config: webpack.Configuration): webpack.Output | undefined  {
+
+  if (!config.output || !config.output.filename) {
+    return config.output;
+  }
+
+  const filename = config.output.filename.replace('[name]', '[name].modern');
+
+  return { ...config.output, filename };
+}
+
+function buildModernResolve(config: webpack.Configuration): webpack.Resolve | undefined {
+
+  if (!config.resolve || !config.resolve.mainFields) {
+    return config.resolve;
+  }
+
+  const modernResolve = { ...config.resolve };
+  modernResolve.mainFields = ['es2015', 'module', 'browser', 'main'];
+
+  return modernResolve;
+}
+
+function buildLegacyResolve(config: webpack.Configuration): webpack.Resolve | undefined {
+
+  if (!config.resolve || !config.resolve.mainFields) {
+    return config.resolve;
+  }
+
+  const modernResolve = { ...config.resolve };
+  modernResolve.mainFields = ['module', 'browser', 'main'];
+
+  return modernResolve;
+}
+
+function buildModernEntry(config: webpack.Configuration): webpack.Entry | undefined {
+  const entry = config.entry;
+
+  if (!entry) {
+    return undefined;
+  }
+
+  const polyfillsEntry = entry as webpack.Entry;
+
+  const polyfills: string[] = polyfillsEntry['polyfills'] as string[];
+
+  if (!polyfills) {
+    return { ...polyfillsEntry };
+  }
+
+  const tweakPolyfills = (file: string) => {
+
+    // TODO: Do we need an own polyfill file for modern browsers? If yes, where is it created?
+
+    if (file.endsWith('polyfills.ts')) {
+      return file.replace('polyfills.ts', 'polyfills.modern.ts');
+    } else {
+      return file;
+    }
+  };
+
+  const modernPolyfills = polyfills.map(tweakPolyfills);
+  const modernEntry = { ...entry as webpack.Entry, polyfills: modernPolyfills };
+
+  return modernEntry;
+}
+
+function buildLegacyPlugins(config: webpack.Configuration): webpack.Plugin[] | undefined {
```

Comment:
Why not generate a webpack config which is already property configured for an es2015 or es5 build?